src/apps/backend/modules/account/account-service.ts

```diff
@@ -22,4 +34,81 @@ export default class AccountService {
   ): Promise<Account> {
     return AccountReader.getAccountById(params.accountId);
   }
+
+  public static async createPasswordResetToken(
+    params: CreatePasswordResetTokenParams,
+  ): Promise<PasswordResetToken> {
+    const account = await AccountReader.getAccountByUsername(params.username);
+
+    const passwordResetToken = AccountUtil.generatePasswordResetToken();
+
+    const passwordResetTokenDbData = await AccountWriter.createPasswordResetToken(
+      account.id,
+      passwordResetToken,
+    );
+
+    await this.sendPasswordResetEmail(
+      account.id,
+      account.firstName,
+      account.username,
+      passwordResetToken,
+    );
+    return passwordResetTokenDbData;
+  }
+
+  public static async resetPassword(
```

Comment:
There should be an existing PATCH call on account, right? If not we should have that i.e we should continue to have a single path to update properties on account, include password.

Reply:
But sir, for resetting password first we need to verify the token that have been sent to the email and the params will be different. Whereas, for other account updates like firstName/lastName we will verify the JWT token in the header using auth middleware.
The reset password route needs to be a public and different PATCH call, as we don't have any auth header because we are resetting the password without login. And I believe 
> we should continue to have a single path to update properties on account, include password.

this is possible when the user is logged in.